File: frontend/src/services/authLetterRecord.ts
import api from './api'

export interface AuthLetterRecord {
  id: number
  project_id: number
  project_name: string
  project_number: string
  round_number: number
  bid_time: string
  supervisor_name: string
  representative_names: string
  generated_by: string
  generated_at: string
}

export interface AuthLetterRecordInput {
  project_id: number
  project_name: string
  project_number: string
  round_number: number
  bid_time: string
  supervisor_name: string
  representative_names: string
}

export const listAuthLetterRecords = () =>
  api.get<{ ok: boolean; data: AuthLetterRecord[] }>('/auth-letter-records')

export const createAuthLetterRecord = (data: AuthLetterRecordInput) =>
  api.post<{ ok: boolean; message: string; data: AuthLetterRecord }>('/auth-letter-records', data)

export const deleteAuthLetterRecord = (id: number) =>
  api.delete<{ ok: boolean; message: string }>(`/auth-letter-records/${id}`)

// 按记录重新生成并下载授权函 Word（经办人/代理机构均可下载自己有权的记录）
export const downloadAuthLetterRecordWord = (id: number) =>
  api.get<Blob>(`/auth-letter-records/${id}/word`, { responseType: 'blob' })

File: backend/models/auth_letter_record.py
from . import db


class AuthLetterRecord(db.Model):
    __tablename__ = "auth_letter_records"

    id = db.Column(db.Integer, primary_key=True)
    project_id = db.Column(db.Integer, nullable=False)
    project_name = db.Column(db.String(200), default="")
    project_number = db.Column(db.String(50), default="")
    round_number = db.Column(db.Integer, default=1)
    bid_time = db.Column(db.String(60), default="")
    supervisor_name = db.Column(db.String(50), default="")
    representative_names = db.Column(db.Text, default="")  # 逗号分隔
    generated_by = db.Column(db.String(50), default="")
    generated_at = db.Column(db.String(30), default="")

    def to_dict(self):
        return {
            "id": self.id,
            "project_id": self.project_id,
            "project_name": self.project_name or "",
            "project_number": self.project_number or "",
            "round_number": self.round_number or 1,
            "bid_time": self.bid_time or "",
            "supervisor_name": self.supervisor_name or "",
            "representative_names": self.representative_names or "",
            "generated_by": self.generated_by or "",
            "generated_at": self.generated_at or "",
        }
